reference.md

```diff
@@ -102,8 +102,12 @@ for loop
     See also: [while loop](#while-loop).
 
 function
-:   A group of instructions (i.e., lines of code) that transform
-    some input [arguments](#argument) to some output.
+:   A named group of instructions that is executed when the function's name is used in
+    the code. Occurrence of a function name in the code is a [function call](#function-call).
+    Functions may process input [arguments](#argument) and return some value(s). If functions
+    are used just to group logical pieces of code together, without a [return statement](#return-statement)
```

Comment:
Suggestion: `return some value(s)` -> `return the result back`.

What do you think about changing "If functions are used..." sentence to:

> Functions may also be used for grouping together logical pieces of code. In such cases, they don't need to return any meaningful value and can be written without the return statement completely. Such functions return a special value `None`, which is a way of saying "nothing" in Python.

Reply:
Yes, I think your suggestions make the language less technical and more understandable.